rclcpp/include/rclcpp/experimental/subscription_intra_process.hpp

```diff
@@ -115,13 +116,34 @@ class SubscriptionIntraProcess : public SubscriptionIntraProcessBase
   bool
   is_ready(rcl_wait_set_t * wait_set)
   {
-    (void)wait_set;
+    (void) wait_set;
     return buffer_->has_data();
   }
 
-  void execute()
+  void take_data(std::shared_ptr<void> & data)
```

Comment:
I'm wondering if this signature should just be:

```
std::shared_ptr<void> take_data()
```

It makes a little more sense to me.

Reply:
Yeah, that sounds like the correct signature to me.
@audrow do you think that change would be ok?

Reply:
Done in https://github.com/ros2/rclcpp/pull/1241/commits/bc0849daa4b326cf7a5f46c575452c775d408782